### PR TITLE
feat(ops): handle real_ip_from CIDR format

### PR DESCRIPTION
**Pull request by kwanhur**
### Description
Handle `real_ip_from` CIDR format on plugin batch-requests when APISIX init.

* Add `cli.ip` module with lua-resty-mediador/ip.lua to verify
* Support loopback and unspecified(zeros) ip address
* Support IPv4 and IPv6

Fixes # (issue)
https://github.com/apache/apisix/blob/cc88a8db0ba64e1cb351e389ca5ec64c516073d6/apisix/cli/ops.lua#L279-L281

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Comment by kwanhur** [comment]
`cjson` not in `apisix-master-0.rockspec`, any missing it? @spacewander 

Using `apisix/core/ip.lua`, here require `apisix/core/json` where original `cjson` reference.

Test case failed cause at `cjson`

```plaintext
/apisix/bin/apisix init
/usr/local/openresty-debug/luajit/bin/luajit ./apisix/cli/apisix.lua init
/usr/local/openresty-debug/luajit/bin/luajit: ./apisix/core/json.lua:22: module 'cjson.safe' not found:
	no field package.preload['cjson.safe']
	no file '/apisix/cjson/safe/init.lua'
	no file '/apisix/deps/share/lua/5.1/cjson/safe/init.lua'
	no file '/apisix/deps/share/lua/5.1/cjson/safe.lua'
	no file '/usr/local/apisix/deps/share/lua/5.1/cjson/safe.lua'
	no file './cjson/safe.lua'
```

**Comment by kwanhur** [comment]
I found the `cjson.so` place on `/usr/local/openresty-debug/lualib/` directory, but not it the `package.cpath`.

**Comment by kwanhur** [comment]
I need some help @tokers @tzssangglass

**Comment by tzssangglass** [comment]
A bit tricky.

As I recall cjson is only loaded after APISIX is run (it belongs to the OpenResty runtime). During the startup phase, the cli module, you have to reference three functions of the ip module, one of which will necessarily load json.

**Comment by tzssangglass** [comment]
I think we can't use the `create_ip_matcher` function of the `ip` module until apisix is running.

**Comment by tzssangglass** [comment]
> What about `djson` instead of `cjson`?

just try

**Comment by kwanhur** [comment]
After a trial, `lua-resty-ipmatcher` not good on `apisix.cli.ops` because it requires `lua-resty-core` runtime :-(

[lua-libcidr-ffi](https://github.com/GUI/lua-libcidr-ffi) maybe a choice, it bases on [libcidr](http://www.over-yonder.net/~fullermd/projects/libcidr) a C library.

File: t/cli/test_validate_config.sh [hunk 1]
```diff
@@ -167,11 +167,79 @@ plugins:
...
+        - "127.0.0.1"
+' > conf/config.yaml
+
+out=$(make init 2>&1 || true)
+if echo "$out" | grep "missing loopback or unspecified in the nginx_config.http.real_ip_from for plugin batch-requests"; then
+    echo "failed: should check the realip configuration for batch-requests"
+    exit 1
+fi
+
+echo "passed: check the realip configuration for batch-requests"
```
Comment (spacewander) [inline t/cli/test_validate_config.sh:196]: For multi checks we only need to put one `passed` at the end, like:
https://github.com/apache/apisix/blob/0d1280cde7fcd463f2e86941935e053561bcf331/t/cli/test_validate_config.sh#L149-L162

**Review by spacewander — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**